### PR TITLE
Add extensions build, and a container images upload task

### PR DESCRIPTION
**Pull request by sherine-k**

### CODE REVIEW

**Comment by lmzuccarelli** [comment]
@sherine-k - I'll take a look and try run this locally for verification

File: manifests/tekton/tasks/base/cosa-upload-containers.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        requests:
+          devices.kubevirt.io/kvm: '1'
+          memory: 4Gi
+      script: |
+        #!/usr/bin/env bash
+        set -euxo pipefail
+
+        cd /workspace/coreos
+        cosa build-extensions-container
+        cosa push-container --image=extensions-container  --authfile=/tmp/dockerconfig.json  quay.io/okd/centos-stream coreos-9:devel
```
Comment (LorbusChris) [inline manifests/tekton/tasks/base/cosa-upload-containers.yaml]: For now, we can push just the base container, and buid and push the extensions later once https://github.com/openshift/os/pull/991 has merged.
IIRC the image tag that is pushed to should be generated automatically from the build version
```suggestion
        #cosa build-extensions-container
        cosa push-container quay.io/okd/centos-stream-coreos-9
        #cosa push-container --image=extensions-container quay.io/okd/centos-stream-coreos-9-extensions
```

**Comment by sherine-k** [comment]
/hold
waiting for https://github.com/openshift/os/pull/991 to be merged first

File: manifests/tekton/tasks/base/cosa-upload-containers.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          memory: 4Gi
+        requests:
+          devices.kubevirt.io/kvm: '1'
+          memory: 4Gi
+      script: |
+        #!/usr/bin/env bash
+        set -euxo pipefail
+
+        cd /workspace/coreos
+        cosa build-extensions-container
```
Comment (LorbusChris) [inline manifests/tekton/tasks/base/cosa-upload-containers.yaml]: @sherine-k I think we can just drop `cosa build-extensions-container` for now, and only do the upload of the base container here

File: manifests/tekton/tasks/base/cosa-upload-containers.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        cosa push-container --image=extensions-container  $(params.target-registry) coreos-9:$(params.target-image-tag)
```
Comment (LorbusChris) [inline manifests/tekton/tasks/base/cosa-upload-containers.yaml]:
```suggestion
        cosa push-container $(params.target-registry) coreos-9:$(params.target-image-tag)
```

File: manifests/tekton/tasks/base/kustomization.yaml [hunk 4]
```diff
@@ -2,5 +2,6 @@ bases:
   - cosa-init.yaml
   - cosa-build.yaml
   - cosa-buildextend.yaml
+  - cosa-upload-containers.yaml
```
Comment (LorbusChris) [inline manifests/tekton/tasks/base/kustomization.yaml]: let's call this task and the file `cosa-upload-images`

File: manifests/tekton/tasks/base/cosa-upload-containers.yaml (hunk 1 above)
Comment (sherine-k) [reply manifests/tekton/tasks/base/cosa-upload-containers.yaml]: OK, kept only push-container, base container

File: manifests/tekton/tasks/base/cosa-upload-containers.yaml (hunk 2 above)
Comment (sherine-k) [reply manifests/tekton/tasks/base/cosa-upload-containers.yaml]: Done. Sorry, missed the previous comment

File: manifests/tekton/tasks/base/kustomization.yaml (hunk 4 above)
Comment (sherine-k) [reply manifests/tekton/tasks/base/kustomization.yaml]: Task renamed and moved.
Thanks @LorbusChris

File: environments/overlays/local/pipelineruns/okd-coreos-all-pipelinerun.yaml [hunk 5]
```diff
@@ -14,8 +14,15 @@ spec:
       value: "4.12"
     - name: rpm-artifacts-image
       value: "registry.ci.openshift.org/origin/4.12:artifacts"
+    - name: target-registry
+      value: "quay.io/okd/centos-stream"
```
Comment (LorbusChris) [inline environments/overlays/local/pipelineruns/okd-coreos-all-pipelinerun.yaml]:
```suggestion
      value: "quay.io/okd/centos-stream-coreos-9"
```

File: manifests/tekton/tasks/base/cosa-upload-images.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+          memory: 4Gi
+        requests:
+          devices.kubevirt.io/kvm: '1'
+          memory: 4Gi
+      script: |
+        #!/usr/bin/env bash
+        set -euxo pipefail
+
+        cd /workspace/coreos
+        cosa push-container  $(params.target-registry) coreos-9:$(params.target-image-tag)
```
Comment (LorbusChris) [inline manifests/tekton/tasks/base/cosa-upload-images.yaml]:
```suggestion
        cosa push-container  $(params.target-registry):$(params.target-image-tag)
```

**Review by LorbusChris — COMMENTED** [review]
Thank for adding the details to the README @sherine-k.
A couple of nits

File: manifests/tekton/tasks/base/cosa-upload-images.yaml (hunk 6 above)
Comment (sherine-k) [reply manifests/tekton/tasks/base/cosa-upload-images.yaml]: With @lmzuccarelli , we had a doubt that this didnt look like a regular image format :+1:

File: environments/overlays/local/pipelineruns/okd-coreos-all-pipelinerun.yaml (hunk 5 above)
Comment (sherine-k) [reply environments/overlays/local/pipelineruns/okd-coreos-all-pipelinerun.yaml]: Done

File: manifests/tekton/pipelines/base/okd-coreos-all.yaml [hunk 7]
```diff
@@ -19,6 +19,12 @@ spec:
     - default: "registry.ci.openshift.org/origin/4.12:artifacts"
       name: rpm-artifacts-image
       type: string
+    - name: target-registry
+      default: "quay.io/okd/centos-stream"
```
Comment (LorbusChris) [inline manifests/tekton/pipelines/base/okd-coreos-all.yaml]:
```suggestion
      default: "quay.io/okd/centos-stream-coreos-9"
```

File: environments/overlays/operate-first/pipelineruns/okd-coreos-all-pipelinerun.yaml [hunk 8]
```diff
@@ -14,8 +14,15 @@ spec:
       value: "4.12"
     - name: rpm-artifacts-image
       value: "registry.ci.openshift.org/origin/4.12:artifacts"
+    - name: target-registry
+      value: "quay.io/okd/centos-stream"
```
Comment (LorbusChris) [inline environments/overlays/operate-first/pipelineruns/okd-coreos-all-pipelinerun.yaml]:
```suggestion
      value: "quay.io/okd/centos-stream-coreos-9"
```

File: manifests/tekton/tasks/base/cosa-upload-images.yaml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+          memory: 4Gi
+        requests:
+          devices.kubevirt.io/kvm: '1'
+          memory: 4Gi
+      script: |
+        #!/usr/bin/env bash
+        set -euxo pipefail
+
+        cd /workspace/coreos
+        cosa push-container  $(params.target-registry):$(params.target-image-tag)
```
Comment (LorbusChris) [inline manifests/tekton/tasks/base/cosa-upload-images.yaml]:
```suggestion
        cosa push-container $(params.target-registry):$(params.target-image-tag)
```
tiny nit

**Review by LorbusChris — CHANGES_REQUESTED** [review]
I think I missed those earlier. I think we're almost there :grimacing:

File: environments/overlays/operate-first/pipelineruns/okd-coreos-all-pipelinerun.yaml (hunk 8 above)
Comment (sherine-k) [reply environments/overlays/operate-first/pipelineruns/okd-coreos-all-pipelinerun.yaml]: Ouch, yes, absolutely, sorry I missed it

File: manifests/tekton/pipelines/base/okd-coreos-all.yaml [hunk 10]
```diff
@@ -66,9 +72,23 @@ spec:
       workspaces:
         - name: ws
           workspace: shared-workspace
-    - name: cosa-buildextend
+    - name: cosa-upload-containers
```
Comment (LorbusChris) [inline manifests/tekton/pipelines/base/okd-coreos-all.yaml]:
```suggestion
    - name: cosa-upload-images
```

File: manifests/tekton/pipelines/base/okd-coreos-all.yaml [hunk 11]
```diff
@@ -66,9 +72,23 @@ spec:
...
+        - name: target-registry
+          value: $(params.target-registry)
+        - name: target-image-tag
+          value: $(params.target-tag)
+      workspaces:
+        - name: ws
+          workspace: shared-workspace
+    - name: cosa-buildextend
+      runAfter:
+        - cosa-upload-containers
```
Comment (LorbusChris) [inline manifests/tekton/pipelines/base/okd-coreos-all.yaml]:
```suggestion
        - cosa-upload-images
```

**Review by LorbusChris — CHANGES_REQUESTED** [review]

**Review by LorbusChris — APPROVED** [review]

**Comment by LorbusChris** [comment]
Tested locally, and even pushed the first base and extensions images to quay :tada:

@sherine-k @lmzuccarelli please take another look and merge at will

**Pull request merged**